src/Spryker/Client/RabbitMq/Model/Connection/ConnectionBuilder/ConnectionBuilder.php

```diff
@@ -69,8 +69,10 @@ public function createConnectionByQueueConnectionTransfer(QueueConnectionTransfe
      */
     protected function createOrGetConnection(QueueConnectionTransfer $queueConnectionTransfer): ConnectionInterface
     {
-        if (isset($this->createdConnectionsByConnectionName[$queueConnectionTransfer->getName()])) {
-            return $this->createdConnectionsByConnectionName[$queueConnectionTransfer->getName()];
+        $connection = $this->createdConnectionsByConnectionName[$queueConnectionTransfer->getName()] ?? null;
+
+        if ($connection !== null && $connection->getChannel()->is_open()) {
```

Comment:
Are we allowed to write `$connection?->getChannel()` ?

Reply:
we are according to php version (>=8.1)but it is not obvious boolean and can return null 

I would leave it for CORE review if it is allowed